moveit_ros/planning/kinematics_plugin_loader/src/kinematics_plugin_loader.cpp

```diff
@@ -293,27 +303,26 @@ moveit::core::SolverAllocatorFn KinematicsPluginLoader::getLoaderFunction(const
       if (default_solver_plugin_.empty())
       {
         RCLCPP_DEBUG(LOGGER, "Loading settings for kinematics solvers from the ROS param server ...");
-
-        // TODO(JafarAbdi): revise parameter lookup
         // read the list of plugin names for possible kinematics solvers
         for (const srdf::Model::Group& known_group : known_groups)
         {
           std::string base_param_name = known_group.name_;
           std::string ksolver_param_name = base_param_name + ".kinematics_solver";
           RCLCPP_DEBUG(LOGGER, "Looking for param %s ", ksolver_param_name.c_str());
-          bool found = node_->has_parameter(ksolver_param_name);
-          if (!found)
+          rclcpp::Parameter ksolver_param = declare_parameter(node_, ksolver_param_name);
```

Comment:
Why do we need these additional checks? Are we running multiple IK solver instances in the same node namespace? Otherwise, just calling `Node::declare_parameter()` should be sufficient.